bootloader/src/pyi_launch.c

```diff
@@ -358,6 +358,46 @@ pyi_launch_extract_binaries(ARCHIVE_STATUS *archive_status)
     return retcode;
 }
 
+// #if defined(LAUNCH_DEBUG) && defined(WINDOWED)
+static void 
+_handle_system_exit(PyObject *exception, PyObject *value, PyObject *tb) {
+    int exitcode = 0;
+    
+    FATALERROR("fetching exception");
+    PI_PyErr_Fetch(&exception, &value, &tb);
+    FATALERROR("exception fetched");
+    if (!PI_PyErr_GivenExceptionMatches(exception, PI_PyExc_SystemExit)) {
+        FATALERROR("Exception does not match SystemExit");
+        return;
+    }
+    FATALERROR("The exception is a SystemExit");
+
+    if (value == NULL || value == PI_Py_None) {
+        FATALERROR("Exiting with code: %i\n", exitcode);
+        goto done;
+    }
+    if (PI_PyLong_Check(value)) {
+        exitcode = (int) PI_PyLong_AsLong(value);
+        FATALERROR("Exiting with code: %i\n", exitcode);
+    } else {
+        PyObject* code = PI_PyObject_GetAttrString(value, "code");
+        exitcode = (int) PI_PyLong_AsLong(code);
```

Comment:
What is this case? Please add a short comment.
This will fail if value doe not have an attribute "code".